src/base/CMakeLists.txt

```diff
@@ -1,13 +1,16 @@
+add_library(random OBJECT random.cpp)
```

Comment:
This this really nessecary? I think it could be removed

Reply:
It's a more fine-grained control: most of the library depends only on boost headers, where this particular file also depends on math, which is statically linked.
Basically it shows where the compilation bottleneck is, otherwise we could have just used header-only boost without any compilation.